plugins/modules/gpg_keypair.py

```diff
@@ -0,0 +1,390 @@
+        type: str
+        default: present
+        choices: [ present, absent ]
+    key_type:
+        description:
+            - "Specifies the type of key to create. By default this is V(EDDSSA) which must be used with Curve25519.
+              Supported key types are V(RSA), V(DSA), V(ECDSA), V(EDDSA), and V(ECDH)."
+        type: str
+        default: EDDSA
+        choices: ['RSA', 'DSA', 'ECDSA', 'EDDSA', 'ECDH']        
+    key_length:
+        description:
+            - For non-ECC keys, this specifies the number of bits in the key to create.
+            - For RSA keys, the minimum is V(1024), the maximum is V(4096), and the default is V(3072).
+            - For DSA keys, the minimum is V(768), the maximum is V(3072), and the default is V(2048).
+            - Invalid values will automatically be saturated in the afforemented ranges for each respective key.
+            - For ECC keys, this parameter will be ignored.
+        type: int
+    key_curve:
+        description:
+            - For ECC keys, this specifies the curve used to generate the keys.
+            - Supported key curves are V(cv25519), V(nistp256), V(nistp384), V(nistp521), V(brainpoolP256r1), V(brainpoolP384r1), V(brainpoolP512r1), and V(secp256k1).
+            - EDDSA keys can only be used with V(cv25519).
+            - Only EDDSA and ECDH keys support V(cv25519), and for both, V(cv25519) is the default.
+            - For ECDSA and ECDH, the default is V(brainpoolP512r1).
+            - For non-ECC keys, this parameter with be ignored.
+        type: str
+        choices: ['cv25519', 'nistp256', 'nistp384', 'nistp521', 'brainpoolP256r1', 'brainpoolP384r1', 'brainpoolP512r1', 'secp256k1']
+    key_usage:
+        description:
+            - Specifies usage(s) for key.
+            - Support usages are V(encrypt), V(sign), V(auth), V(cert).
+            - V(cert) is given to all primary keys regardess, however can be used to only give V(vert) usage to a key.
+            - If not usage is specified, the valid usages for the given key type with be assigned.
+            - If O(state) is V(absent), this parameter is ignored. 
+        type: list[str]
+        choices: ['encrypt', 'sign', 'auth', 'cert']
+    subkey_type:
+        description:
+            - Similar to O(key_type), but also supports V(ELG).
+        type: str
+        default: EDDSA
+        choices: ['RSA', 'DSA', 'ECDSA', 'EDDSA', 'ECDH', 'ELG']
+    subkey_length:
+        description:
+            - Similar to O(key_length).
+            - For ELG keys, the minimum is V(1024), the maximum is V(4096), and the default is V(3072)."
+        type: int
+    subkey_curve:
+        description:
+            - "Similar to O(key_curve)"
+        type: str
+        choices: ['cv25519', 'nistp256', 'nistp384', 'nistp521', 'brainpoolP256r1', 'brainpoolP384r1', 'brainpoolP512r1', 'secp256k1']
+    key_usage:
```

Comment:
Should this be `subkey_usage`?

Reply:
Yes, it should be subkey_usage. Thanks for catching that.